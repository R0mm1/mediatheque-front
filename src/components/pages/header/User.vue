<template>
    <div id="headerUserElement">
        <InputButton :name="'userAccount'" :label-custom-classes="'fas fa-user-circle'"
                     v-on:click.native="goToAccount"/>
    </div>
</template>

<script>
    import InputButton from "../../form/elements/InputButton";

    export default {
        name: "User",
        components: {InputButton},
        methods: {
            goToAccount: function () {
                window.location = '/account'
            }
        }
    }
</script>

<style lang="scss">
    #headerUserElement {
        padding: 10px 40px;
        display: flex;
        flex-direction: column;
        justify-content: center;

        .form_element_button {
            $dimension: 30px;

            border-radius: 50%;
            height: $dimension;
            width: $dimension;
            text-align: center;
            display: flex;
            flex-direction: column;
            justify-content: center;
            font-size: 25px;
            color: #2b2b2b;

            label {
                height: $dimension;
                line-height: $dimension;
                width: initial !important;
                margin-left: 0 !important;
            }
        }
    }
</style>
