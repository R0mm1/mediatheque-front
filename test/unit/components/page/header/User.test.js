import {mount} from '@vue/test-utils'
import User from "@/components/pages/header/User.vue";

describe('User', () => {
    const wrapper = mount(User);

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});
