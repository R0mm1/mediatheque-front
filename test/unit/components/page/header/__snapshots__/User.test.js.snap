// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correctly 1`] = `
<div
  id="headerUserElement"
>
  <div
    class="form_element form_element_button style_normal"
  >
    <label
      class="element_label fas fa-user-circle"
      for="userAccount"
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name="userAccount"
        type="button"
      />
    </div>
  </div>
</div>
`;
